File: examples/client-gen/tictactoe/errors/custom.py
import typing
from anchorpy.error import ProgramError


class TileOutOfBounds(ProgramError):
    def __init__(self) -> None:
        super().__init__(6000, None)

    code = 6000
    name = "TileOutOfBounds"
    msg = None


class TileAlreadySet(ProgramError):
    def __init__(self) -> None:
        super().__init__(6001, None)

    code = 6001
    name = "TileAlreadySet"
    msg = None


class GameAlreadyOver(ProgramError):
    def __init__(self) -> None:
        super().__init__(6002, None)

    code = 6002
    name = "GameAlreadyOver"
    msg = None


class NotPlayersTurn(ProgramError):
    def __init__(self) -> None:
        super().__init__(6003, None)

    code = 6003
    name = "NotPlayersTurn"
    msg = None


class GameAlreadyStarted(ProgramError):
    def __init__(self) -> None:
        super().__init__(6004, None)

    code = 6004
    name = "GameAlreadyStarted"
    msg = None


CustomError = typing.Union[
    TileOutOfBounds, TileAlreadySet, GameAlreadyOver, NotPlayersTurn, GameAlreadyStarted
]
CUSTOM_ERROR_MAP: dict[int, CustomError] = {
    6000: TileOutOfBounds(),
    6001: TileAlreadySet(),
    6002: GameAlreadyOver(),
    6003: NotPlayersTurn(),
    6004: GameAlreadyStarted(),
}


def from_code(code: int) -> typing.Optional[CustomError]:
    maybe_err = CUSTOM_ERROR_MAP.get(code)
    if maybe_err is None:
        return None
    return maybe_err

File: ts-reference/examples/tic-tac-toe/generated-client/errors/custom.ts
export type CustomError =
  | TileOutOfBounds
  | TileAlreadySet
  | GameAlreadyOver
  | NotPlayersTurn
  | GameAlreadyStarted

export class TileOutOfBounds extends Error {
  readonly code = 6000
  readonly name = "TileOutOfBounds"
  readonly msg = "undefined"

  constructor() {
    super("6000: undefined")
  }
}

export class TileAlreadySet extends Error {
  readonly code = 6001
  readonly name = "TileAlreadySet"
  readonly msg = "undefined"

  constructor() {
    super("6001: undefined")
  }
}

export class GameAlreadyOver extends Error {
  readonly code = 6002
  readonly name = "GameAlreadyOver"
  readonly msg = "undefined"

  constructor() {
    super("6002: undefined")
  }
}

export class NotPlayersTurn extends Error {
  readonly code = 6003
  readonly name = "NotPlayersTurn"
  readonly msg = "undefined"

  constructor() {
    super("6003: undefined")
  }
}

export class GameAlreadyStarted extends Error {
  readonly code = 6004
  readonly name = "GameAlreadyStarted"
  readonly msg = "undefined"

  constructor() {
    super("6004: undefined")
  }
}

export function fromCode(code: number): CustomError | null {
  switch (code) {
    case 6000:
      return new TileOutOfBounds()
    case 6001:
      return new TileAlreadySet()
    case 6002:
      return new GameAlreadyOver()
    case 6003:
      return new NotPlayersTurn()
    case 6004:
      return new GameAlreadyStarted()
  }

  return null
}
